code/io/timer.cpp

```diff
@@ -104,7 +106,12 @@ std::uint64_t timer_get_nanoseconds()
 {
 	auto time = get_performance_counter();
```

Comment:
This is the number of ns since the game started. With 1h you have 3 600 000 000 000, so you need 2^44 bits. you will overflow if you multiply by something greater than 2^20, this is a problem